libkbfs/disk_block_cache.go

```diff
@@ -896,35 +902,52 @@ func (cache *DiskBlockCacheLocal) evictLocked(ctx context.Context,
 		}
 	}()
 	numElements := numBlocks * evictionConsiderationFactor
-	blockID, err := cache.getRandomBlockID(numElements, cache.numBlocks)
-	if err != nil {
-		return 0, 0, err
-	}
-	rng := &util.Range{Start: blockID.Bytes(), Limit: cache.maxBlockID}
-	iter := cache.metaDb.NewIterator(rng, nil)
-	defer iter.Release()
-
 	blockIDs := make(blockIDsByTime, 0, numElements)
-
-	for i := 0; i < numElements; i++ {
-		if !iter.Next() {
-			break
+	maxPriorityForEviction := 0
+
+	// Repeatedly double the search space until there are enough evictable
+	// blocks. If we reach the whole cache size, then give up and relax which
+	// blocks can be evicted.
```

Comment:
I'm worried that this will be woefully slow in the situations that we're trying to cover. Especially since we have to iterate through `numElements` each time. In the worst case, we're talking iterating through 2*n elements per max priority. Additionally, you have one loop here iterating over 2 iteration variables: 1 for the `numElements` doubling, and 1 for `maxPriorityForEviction`. So we'll actually iterate over 2*n elements the first time, then n elements 2 more times, 1 per priority.

The pathological case might be quite common--I'm assuming people's caches will be mostly taken up by their own TLF's blocks.

The worst part is: this will happen every time an eviction is attempted, so once you're in the pathological scenario, you'll get slow evictions all the time.

Reply:
I'm to blame for the initial idea for the algorithm.  It was just the easiest, fastest thing I could think of.

Maybe we can we use `cache.tlfDb` to pick candidate keys that aren't in the home TLFs?  That should be pretty fast and straightforward.

I want to avoid re-doing the cache layout if possible, since that sounds hard.

Reply:
Cool.  Sorry, I should have read your overall comment more closely.

I think it should be pretty easy.  We can just make a tlf key with a random ID + the zero block key, and see what the next TLF is, skipping the home directories.  Do that `numElements` times. 
 And we can use `cache.tlfCounts` to check ahead of time if the only TLFs we know about are the home dirs, so we don't get stuck doing a bunch of skip-aheads for no reason.

Reply:
Yes, the `cache.tlfCounts` solution is the right way to go, I think.